source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc

```diff
@@ -157,6 +159,9 @@ class GetAddrInfoDnsResolver : public DnsResolver, public Logger::Loggable<Logge
 
         next_query = std::move(pending_queries_.front());
         pending_queries_.pop_front();
+        if (reresolve && next_query->cancelled_) {
```

Comment:
nit: it's probably not important, but do we need the `reresolve` condition here?  In other words, shouldn't any `next_query` that is in the cancelled state be skipped?  I realize the only way this can happen right now is if reresolve is true below and we get EAI_AGAIN so we add it back to `pending_queries_`.

Reply:
it's currently possible to start a resolve and cancel before it gets popped from the queue at which point we'll do the resolve anyway.  doubt it matters but wanted it guarded just in case.